### PR TITLE
Consolidade documentation in RTD

### PR DESCRIPTION
**Pull request by asilveir**
- Add a FAQ documentation (source: internal doc)
- Add architecture document (source: archived git repo)
- Updated workflow in SVG format.
- Update actor tutorial to add workflow diagram references.
- Move best-pratice from leapp-repository to RTD and update references.
- Move contrib guide from leapp-repository.
- Move leapp-repository dependencies documentation.
- Move inhibit upgrade documentation to RTD.
- Add dependencies documentation to menu.

### CODE REVIEW

File: docs/source/architecture.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+![Architecture overview](https://github.com/leapp-to/leapp-to.github.io/raw/master/img/framework-arch-overview.png)
+
+There are two tools for working with the framework, the end user application `leapp` and the development utility `snactor`. The `leapp` tool is designed to run specific workflows, while the `snactor` tool can run arbitrary workflows, but also individual actors.
+
+A *workflow* describes what work is going to be done and when. Each workflow is made of a sequence of *phases*, which contain *actors* split into three stages - before, main, and after. Workflows, actors, and all the parts necessary for the execution are loaded from repositories.
+
+Each actor is executed in a forked child process to prevent the modification of the application state. All messages and logs produced by the actors are stored in the *audit database*.
+
+For more information about each part of the architecture, check the [terminology](/terminology.html).
```
Comment (examon) [inline docs/source/architecture.md]: Shouldn't the `terminology.html` be absolute path to the `terminology.md`?

File: docs/source/best-practices.md [hunk 2]
```diff
@@ -1,4 +1,153 @@
 # Best Practices for actor developmemt
 
-Read the best practices for actor development located
-[here](https://github.com/oamg/leapp-repository/blob/master/docs/best-practices.md).
+## Follow the contribution guidelines
+
+See the [contribution guidelines for leapp-repository](/contributing).
```
Comment (examon) [inline docs/source/best-practices.md]: same as with terminology

File: docs/source/best-practices.md [hunk 3]
```diff
@@ -1,4 +1,153 @@
...
+by the leapp tool, and also to avoid unnecessary complications for our tests automation which needs to inspect the
+actors beforehand.
+
+## Use the snactor tool for development
+
+The snactor tool helps you with creating the base layout of a new Leapp repository, and with creating boilerplates for
+the repository artifacts like actors, models, tags, topics, and workflows. It also helps you with debugging as it is
+able to execute individual actors.
+
+See the [tutorial on basic usage of snactor](/first-actor).
```
Comment (examon) [inline docs/source/best-practices.md]: same as with terminology

Comment (examon) [reply docs/source/best-practices.md]: It seems like all references to markdown files require full file path.

**Comment by jmikovic** [comment]
none of the links to *.md files are working

**Comment by examon** [comment]
Sorry about that @asilveir . Jozef is right. Having links reference `.md` files does not work.
The correct way should be this.

Instead of the full path to the html,
`For further information about how create an actor read this [document](https://leapp.readthedocs.io/en/latest/first-actor.html)`

please use this approach:
`For further information about how create an actor read this [document](first-actor.html)`

Note: this does not work:
`For further information about how create an actor read this [document](first-actor.md)`

I was testing this via building doc:

```
cd docs
make livehtml
```

**Comment by vojtechsokol** [comment]
leapp-ci build

**Pull request merged**